pkg/controller/jobframework/reconciler.go

```diff
@@ -698,7 +709,8 @@ func equivalentToWorkload(ctx context.Context, c client.Client, job GenericJob,
 	jobPodSets := clearMinCountsIfFeatureDisabled(job.PodSets())
 
 	if runningPodSets := expectedRunningPodSets(ctx, c, wl); runningPodSets != nil {
-		if equality.ComparePodSetSlices(jobPodSets, runningPodSets) {
+		jobSizeable, implementsSizable := job.(ReizableJobs)
+		if equality.ComparePodSetSlices(jobPodSets, runningPodSets) || (implementsSizable && jobSizeable.IsResizable(wl)) {
```

Comment:
This check is not enough.
Since we don't support upsizing, we should return false if upsizing happens here.

But overall, we should check that the shapes are the same (only counters changed).

Reply:
It looks like you are partially checking that inside `IsResizable`, but we shouldn't be doing that in the integration.

It can be generic... just by looking at the podsets.

So, in this package, there could be a function `isDownsized(oldPodsets, newPodSets []PodSet) bool`